build-logic/documentation/src/main/groovy/gradlebuild/docs/GradleJavadocsPlugin.java

```diff
@@ -103,6 +105,23 @@ private void generateJavadocs(Project project, ProjectLayout layout, TaskContain
             if (BuildEnvironment.INSTANCE.getJavaVersion().isJava11Compatible()) {
                 options.addBooleanOption("html4", true);
                 options.addBooleanOption("-no-module-directories", true);
+
+                FileSystemOperations fs = getFs();
+                //noinspection Convert2Lambda
+                task.doLast(new Action<Task>() {
+                    @Override
+                    public void execute(Task task) {
+                        fs.copy(copySpec -> {
+                            // This is a work-around for https://bugs.openjdk.java.net/browse/JDK-8211194. Can be removed once that issue is fixed on JDK's side
+                            // Since JDK 11, package-list is missing from javadoc output files and superseded by element-list file, but a lot of external tools still need it
+                            // Here we generate this file manually
+                            copySpec.from(generatedJavadocDirectory.file("element-list"), sub -> {
+                                sub.rename(t -> "package-list");
+                            });
+                            copySpec.into(generatedJavadocDirectory);
+                        });
+                    }
+                });
```

Comment:
:grey_question:  I'm just interested, but you probably have a good reason: why not a Copy task?

Reply:
We could do that, but we would need to copy all of the javadoc output to another location and make sure everything downstream also looks in that location.  This is correcting the output for the Javadoc task in place.